Updating notices across docs (#4722)
fishtown-analytics/docs.getdbt.com: website/snippets/_cloud-environments-info.md

| Default to Custom Branch | ☑️ | Determines whether to use a branch other than the repository’s default  | See below |
| Custom Branch | dev | Custom Branch name | See below |

### About versions in dbt Cloud

> dbt Cloud allows users to select from multiple dbt releases. At this time, environments must use a dbt version greater than or equal to v1.0.0 [lower versions are no longer supported](/docs/dbt-versions/upgrade-core-in-cloud).

Reviewer: there's a lot of info here and it's pretty important so since we're here, let's turn this into bullets to make it easier for users to read.
```suggestion
- Multiple dbt releases: dbt Cloud enables users to choose from various dbt versions.
- Version requirements: Environments must use dbt version v1.0.0 or higher. [Lower versions are not supported](/docs/dbt-versions/upgrade-core-in-cloud).
- Automatic updates: Selecting a version labeled `(latest)` ensures your environment automatically installs the latest stable release of that minor version selected.
```

Author: Good call!

> If you select a current version with `(latest)` in the name, your environment will automatically install the latest stable version of the minor version selected.

Reviewer: ```suggestion
```

> ### Git repository caching (Enterprise plans only)

Reviewer: ```suggestion
### Git repository caching
```

Author: Let's use the lifecycle status for this to remove that issue
```suggestion
### Git repository caching <Lifecycle status='Enterprise only' />
```

> 

Reviewer: i agree the 'enterprise only' callout should be higher than it currently is. however, I don't think we should add 'enterprise plan only' as part of the header and URL slug. I don't think we should add descriptive words like this in the header because it'll become part of the URL anchor link and can cause 'page not found' URL issues when we eventually add the plan pill component. 

![Screenshot 2024-01-09 at 11 44 39](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/db4b44c9-e889-4319-8c2e-daa2cd66acb7)

also the image 'Example of the Repository caching option' isn't rendering ?

```suggestion
:::info Available for dbt Cloud Enterprise plans
This feature is exclusively available for dbt Cloud Enterprise plans. [Book a demo](https://www.getdbt.com/contact) to learn more.
:::

```

| Default to Custom Branch | ☑️ | Determines whether to use a branch other than the repository’s default  | See below |
| Custom Branch | dev | Custom Branch name | See below |

### About versions in dbt Cloud

- Multiple dbt releases: dbt Cloud enables users to choose from various dbt versions.
- Version requirements: Environments must use dbt version v1.0.0 or higher. [Lower versions are not supported](/docs/dbt-versions/upgrade-core-in-cloud).
- Automatic updates: Selecting a version labeled `(latest)` ensures your environment automatically installs the latest stable release of that minor version selected.

### Git repository caching 

> 

Author: ```suggestion
:::info Available for dbt Cloud Enterprise plans
This feature is exclusively available for dbt Cloud Enterprise plans. [Book a demo](https://www.getdbt.com/contact) to learn more.
:::
```

> ### Git repository caching <Lifecycle status='Enterprise only' />

Author: ```suggestion
### Git repository caching 
```

| Default to Custom Branch | ☑️ | Determines whether to use a branch other than the repository’s default  | See below |
| Custom Branch | dev | Custom Branch name | See below |

> ### About versions in dbt Cloud

Reviewer 2: love this idea of a new section! 

i'm a little familiar with this content. sugg : 
- make it an H2 instead of an H3
- move this new section to a higher location on the page, above the "Common environment settings" H2 section

- Multiple dbt releases: dbt Cloud enables users to choose from various dbt versions.
- Version requirements: Environments must use dbt version v1.0.0 or higher. [Lower versions are not supported](/docs/dbt-versions/upgrade-core-in-cloud).
- Automatic updates: Selecting a version labeled `(latest)` ensures your environment automatically installs the latest stable release of that minor version selected.

> ### Git repository caching 

Reviewer 2: this section is duplicated. 

list is ordered alphabetically, should be after "Extended attributes" section.

here's the page TOC from [vercel preview](https://docs-getdbt-com-git-notices-dbt-labs.vercel.app/docs/deploy/deploy-environments) :

![Screenshot 2024-01-09 at 2 11 23 PM](https://github.com/dbt-labs/docs.getdbt.com/assets/107218380/55f7f015-0948-49ee-856b-840672e12db6)

Author: Going to revert this whole file. Something funky happened with the additions here that weren't supposed to.

:::info Available for dbt Cloud Enterprise plans
This feature is exclusively available for dbt Cloud Enterprise plans. [Book a demo](https://www.getdbt.com/contact) to learn more.

> :::

Reviewer 2: sugg moving this callout to the end of the section. i've been taught that a callout after a title is not "good design" practice

Author: We do this in numerous places. I thought the idea was to give the important info at the top of the article, rather than at the tail end, especially if the feature is not available to a subset of users.

@runleonarun  and @mirnawong1 thoughts on this?

Reviewer: this is a great question and all great points! i would prefer having the callout out at the top of the content:
- we do this already in pretty much all our docs so adding it at the end would be more of an antipattern. 
- i think having at the top sets clear expectations and also hopefully perks interests.
- when i first read this, i thought this was for all plans and then was surprised it was for enterprise. i almost missed the callout, in fact. so i do prefer it to be at the top.
- also if we do the lifecycle pill, that will be in the header so it'll be above the content/article anyway

Reviewer 2: yeah, we have prior art for both. 

i like to talk about the cool thing first instead of dropping caveats/limitations at the top. not a big deal tho. non-blocker for me.
